### ISSUE TITLE
[12.0.0+] normalizeEventKey error (key.toLowerCase)

### ISSUE THREAD

**Issue by Mir0ke** [handsontable/handsontable#9562]
### Description
Handsontable 11 was working fine.
But Handsontable 12.01 now has this issue.
There is no treatment for a nullable key.
Forcing toLowerCase is causing this error.

### Steps to reproduce

1. handsontable\shortcuts\utils.mjs
2. Add handsontable to your page.
3. Add an input text field to your page.
4. Just using your mouse, click inside your input text field and select a value based in your web browser history.
5. Because no keys were used, handsontable produces this console error.

### Demo

https://jsfiddle.net/handsoncode/c6rgz08x/

### Your environment
* React wrapper version: 18.1.0
* Handsontable version: 12.0.1
* Browser name and version: Chrome 102
* Operating system: Windows 10

**Comment by AMBudnik** [comment]
Hi @Mir0ke 

the demo does not seem to fit into the description of steps to replicate the issue. Could you please resend the demo and possibly a recording of the screen when the error pops?

**Comment by AMBudnik** [comment]
I was able to recreate the issue using this demo https://jsfiddle.net/0vadLq5j/3/ 

and get the following error

```js
handsontable.full.js:100655 Uncaught TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    at normalizeEventKey (handsontable.full.js:100655:14)
    at onkeyup (handsontable.full.js:133333:51)
```

![issue](https://user-images.githubusercontent.com/10757813/172160767-b94ed52d-d239-4b79-9985-68d1a81985a7.gif)

And it seems that it is a regression introduced in v12.0.0.

v11.1.0 - https://jsfiddle.net/4mo87szb/2/ works well
v12.0.0 - https://jsfiddle.net/4mo87szb/4/ broken

Thank you very much for sharing!

[note to myself: Inform [Zen](https://handsontable.zendesk.com/agent/tickets/29049)]

**Comment by wszymanski** [comment]
As the @AMBudnik shown, there is thrown a `keyup` event for some reason. It is the way how Gecko browsers works (reproduced on Chrome, Opera, Edge, not seen on Firefox). I'm not sure whether it is a desired behaviour. There should be done some research. 

The Shortcut Manager catches every keyboard events and this event doesn't have `event.key` property set, that is why the error is thrown.

**Comment by Mir0ke** [comment]
> @Mir0ke this issue shows a red error in console (which we want to fix), but does it actually break anything for you? If yes, what is it?

It's not breaking anything.
Just the insistent error that's polluting the console everytime.

**Comment by AMBudnik** [comment]
Hi @Mir0ke I have good news. The Handsontable v12.1.0 is live.

Here you can read the full list of changes https://handsontable.com/docs/release-notes/#_12-1-0. 

If you’d have any questions please feel free to let me know.

**Comment by m-shimojo** [comment]
Hi, we found this error occurring again in v14+.

v13.0.0 - https://jsfiddle.net/9ry1ehgf/ works well
v13.1.0 - https://jsfiddle.net/zjhcgvp6/ works well
v14.0.0 - https://jsfiddle.net/r3yfp672/ broken
v14.1.0 - https://jsfiddle.net/r3yfp672/1/ broken

This time the same error occurs twice.

```js
handsontable.full.js:95717 Uncaught TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    at normalizeEventKey (handsontable.full.js:95717:24)
    at HTMLHtmlElement.onkeydownForModKeys (handsontable.full.js:95836:53)

handsontable.full.js:95717 Uncaught TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    at normalizeEventKey (handsontable.full.js:95717:24)
    at HTMLHtmlElement.onkeyupForModKeys (handsontable.full.js:95849:53)
```

**Comment by adrianszymanski89** [comment]
Hi @m-shimojo 

Thank you for reporting this. I can confirm that this issue is back. I'll report it internally.

**Comment by adrianszymanski89** [comment]
Hi @m-shimojo 

Just wanted to let you know that we released Handsontable 14.2 including the fix for this issue. Please check it and let me know if the issue is gone.

**Comment by AMBudnik** [comment]
Hi @Mir0ke @m-shimojo 

I am happy to announce that we solved this issue in the newest version of Handsontable - v14.2.0.

Here is the pull requests that solves it https://github.com/handsontable/handsontable/pull/10795

And in the Release Notes you can read more about the changes for this version https://handsontable.com/docs/javascript-data-grid/release-notes/#_14-2-0

**Comment by m-shimojo** [comment]
Thanks for letting me know, I've confirmed that the error has been resolved in v14.2.0!

**Comment by AMBudnik** [comment]
Great. Thank you for the update.
